src/general/Mutations.cs

```diff
@@ -206,112 +182,78 @@ private void MutateBehaviour(MicrobeSpecies parent, MicrobeSpecies mutated)
             {
-                bool copy = true;
+                OrganelleTemplate organelle = (OrganelleTemplate)parentOrganelle.Clone();
 
-                var organelle = parentOrganelles[i];
-
-                if (parentOrganelles.Count < 2)
+                // Chance to replace or remove if not a nucleus
+                if (organelle.Definition != nucleus)
                 {
-                    // Removing last organelle would be silly
-                }
-                else if (organelle.Definition != nucleus)
-                {
-                    // Chance to replace or remove if not a nucleus
-
-                    if (random.Next(0.0f, 1.0f) < Constants.MUTATION_DELETION_RATE)
+                    if (random.Next(0.0f, 1.0f) < Constants.MUTATION_DELETION_RATE / Math.Sqrt(parentOrganelles.Count))
                     {
-                        copy = false;
+                        // Don't copy over this organelle, removing this one from the new species
+                        continue;
                     }
-                    else if (random.Next(0.0f, 1.0f) < Constants.MUTATION_REPLACEMENT_RATE)
-                    {
-                        copy = false;
 
-                        var replacer = new OrganelleTemplate(GetRandomOrganelle(isBacteria),
+                    if (random.Next(0.0f, 1.0f) < Constants.MUTATION_REPLACEMENT_RATE)
+                    {
+                        organelle = new OrganelleTemplate(GetRandomOrganelle(isBacteria),
                             organelle.Position, organelle.Orientation);
-
-                        // The replacing organelle might not fit at the same position
-                        try
-                        {
-                            organelles.Add(replacer);
-                        }
-                        catch (ArgumentException)
-                        {
-                            // Couldn't replace it
-                            copy = true;
-                        }
                     }
                 }
 
-                if (!copy)
-                    continue;
-
                 // Copy the organelle
                 try
                 {
-                    organelles.Add((OrganelleTemplate)organelle.Clone());
+                    mutatedOrganelles.Add(organelle);
                 }
                 catch (ArgumentException)
                 {
                     // Add the organelle randomly back to the list to make
                     // sure we don't throw it away
-                    AddNewOrganelle(organelles, organelle.Definition);
+                    AddNewOrganelle(mutatedOrganelles, organelle.Definition);
                 }
             }
 
-            // Can add up to 6 new organelles (Which should allow AI to catch up to player more
             // We can insert new organelles at the end of the list
-            if (random.Next(0.0f, 1.0f) < Constants.MUTATION_CREATION_RATE)
+            for (int i = 0; i < 6; i++)
```

Comment:
use preincrement `++i` here

Reply:
I don't care either way, but may I ask why?

Reply:
Mostly just style. Theoretically preincrement doesn't need to do a copy of the variable to return it like postincrement does, but as it's an int it doesn't really matter (this is true for C++ at least), and the optimizer probably removes the extra copy anyway.